src/components/raycaster.js

```diff
@@ -1,112 +1,142 @@
+
+    // Update intersectedEls object first in case event handlers try to inspect it.
+    intersections.forEach(function emitEvents (intersection) {
+      intersectedEls.push(intersection.object.el);
+    });
 
-      // If the intersected object is the cursor itself
-      // or the object is further than the max distance
+    // Emit intersected on intersected entity per intersected entity.
+    intersections.forEach(function emitEvents (intersection) {
+      var intersectedEl = intersection.object.el;
+      intersectedEl.emit('raycaster-intersected', {el: el, intersection: intersection});
+    });
 
-      if (intersectedObj.object.el === undefined) { continue; }
-      if (intersectedObj.object.el === cursorEl) { continue; }
-      if (!intersectedObj.object.visible) { continue; }
-      return intersectedObj;
+    // Emit all intersections at once on raycasting entity.
+    if (intersections.length) {
+      el.emit('raycaster-intersection', {
+        els: intersections.map(function getEl (intersection) {
+          return intersection.object.el;
+        }),
+        intersections: intersections
+      });
     }
-    return null;
-  },
 
-  /**
-   * Remembers the last intersected element
-   */
-  setExistingIntersection: function (el, distance) {
-    this.intersectedEl = el;
-    this.el.emit('intersection', { el: el, distance: distance });
+    // Emit intersection cleared on both entities per formerly intersected entity.
+    prevIntersectedEls.forEach(function checkStillIntersected (intersectedEl) {
+      if (intersectedEls.indexOf(intersectedEl) !== -1) { return; }
+      el.emit('raycaster-intersection-cleared', {el: intersectedEl});
+      intersectedEl.emit('raycaster-intersected-cleared', {el: el});
+    });
   },
 
   /**
-   * Emits a `mouseleave` event and clears info about the last intersection.
+   * Set origin and direction of raycaster using entity position and rotation.
    */
-  clearExistingIntersection: function () {
-    var intersectedEl = this.intersectedEl;
-    this.el.emit('intersectioncleared', { el: intersectedEl });
-    this.intersectedEl = null;
-  },
+  updateOriginDirection: (function () {
+    var directionHelper = new THREE.Quaternion();
+    var originVec3 = new THREE.Vector3();
 
-  handleIntersection: function (obj) {
-    var el = obj.object.el;
+    // Closure to make quaternion/vector3 objects private.
+    return function updateOriginDirection () {
+      var el = this.el;
+      var object3D = el.object3D;
 
-    // A new intersection where previously a different element was
-    // and now needs a mouseleave event.
-    if (this.intersectedEl !== el) {
-      this.clearExistingIntersection();
-    }
-    this.setExistingIntersection(el, obj.distance);
-  }
+      // Update matrix world.
+      object3D.updateMatrixWorld();
+      // Grab the position and rotation.
+      object3D.matrixWorld.decompose(originVec3, directionHelper, scaleDummy);
```

Comment:
I'm not actually using `scaleDummy`. I just pass it because `decompose` will expect it.